griptape/drivers/image_generation/openai_image_generation_driver.py

```diff
@@ -95,7 +118,32 @@ def try_image_inpainting(
         mask: ImageArtifact,
         negative_prompts: Optional[list[str]] = None,
     ) -> ImageArtifact:
-        raise NotImplementedError(f"{self.__class__.__name__} does not support inpainting")
+        if self.model != "dall-e-2":
+            raise NotImplementedError(f"{self.__class__.__name__} only supports dall-e-2 for image variation")
+
+        if self.image_size not in {"256x256", "512x512", "1024x1024"}:
+            raise ValueError(f"dall-e-2 image variation support image sizes 256x256, 512x512, and 1024x1024")
+
+        image_size = cast(Literal["256x256", "512x512", "1024x1024"], self.image_size)
```

Comment:
Annoying stuff happening here: while both `dall-e-2` and `dall-e-3` use the same SDK, only `dall-e-2` supports editing (inpainting) and variations, at least for now. This means we need some checks to determine if the driver configuration makes sense for the method call. 

Maybe we need to split into two distinct model drivers given the distinct behavior?

Reply:
That is annoying, but the Model Driver pattern feels more applicable to platforms that provide models from different vendors. I think these one off checks are fine for now as long as it's minimal.